modules/Menuing.py

```diff
@@ -0,0 +1,584 @@
+import os
+import struct
+from typing import NoReturn
+
+from modules.Config import config_battle
+from modules.Inputs import PressButton, WaitFrames
+from modules.Memory import ReadSymbol, GetGameState, pokemon_list, type_list, GetParty, GetOpponent, DecodeString, \
+    GetPartyMenuCursorPos, ParseStartMenu, ParseMenu, ParseBattleCursor, mGBA, ParseTasks, ReadMemory, \
+    moves_list, ParseMain, GameState, TaskFunc, GetTaskFunc
+from modules.Console import console
+
+if mGBA.game in ['Pokémon Ruby', 'Pokémon Sapphire']:
+    battle_text = 'What should'  # TODO English only
+else:
+    battle_text = 'What will'  # TODO English only
+
+
+def SelectBattleOption(desired_option: int, cursor_type: str = 'gActionSelectionCursor') -> NoReturn:
+    """
+    Takes a desired battle menu option, navigates to it, and presses it.
+
+    :param desired_option: The desired index for the selection. For the base battle menu, 0 will be FIGHT, 1 will be
+    BAG, 2 will be PKMN, and 3 will be RUN.
+    :param cursor_type: The symbol to use for the cursor. This is different between selecting moves and selecting battle
+     options.
+    """
+    while ParseBattleCursor(cursor_type) != desired_option:
+        match (ParseBattleCursor(cursor_type) % 2) - (desired_option % 2):
+            case - 1:
+                PressButton(['Right'])
+            case 1:
+                PressButton(['Left'])
+        match (ParseBattleCursor(cursor_type) // 2) - (desired_option // 2):
+            case - 1:
+                PressButton(['Down'])
+            case 1:
+                PressButton(['Up'])
+            case 0:
+                pass
+    if ParseBattleCursor(cursor_type) == desired_option:
+        # get current displayed string
+        current_string = DecodeString(ReadSymbol('gDisplayedStringBattle'))  # TODO English only
+        # mash A until the string changes
+        while DecodeString(ReadSymbol('gDisplayedStringBattle')) == current_string:  # TODO English only
+            PressButton(['A'])
+
+
+def FleeBattle() -> NoReturn:
+    """
+    Readable function to select and execute the Run option from the battle menu.
+    """
+    while GetGameState() != GameState.OVERWORLD:
+        if 'Use next' in DecodeString(ReadSymbol('gDisplayedStringBattle')):  # TODO English only
```

Comment:
I have tagged all text box/string checks with `# TODO English only` - we definitely need to look at language agnostic options instead 😭